pkg/kubectl/cmd/update.go

```diff
@@ -116,12 +116,13 @@ func updateWithPatch(cmd *cobra.Command, args []string, f *Factory, patch string
 	obj, err := helper.Get(namespace, name)
 	checkErr(err)
 
-	cmdutil.Merge(obj, patch, mapping.Kind)
+	patchedObj, err := cmdutil.Merge(obj, patch, mapping.Kind)
+	checkErr(err)
 
-	data, err := helper.Codec.Encode(obj)
+	data, err := helper.Codec.Encode(patchedObj)
 	checkErr(err)
 
-	obj, err = helper.Update(namespace, name, true, data)
+	_, err = helper.Update(namespace, name, true, data)
```

Comment:
This particular line does not but the PR does. Check out merge_test.go. Just to be sure i've added some test to helper_test.go as well. See
https://github.com/GoogleCloudPlatform/kubernetes/pull/4448/files#diff-c8e282cc368fe38828935844bb3bd00aR113
which delete's a field and
https://github.com/GoogleCloudPlatform/kubernetes/pull/4448/files#diff-c8e282cc368fe38828935844bb3bd00aR133
which updates an int field to zero. default-value updates are fixed by applying patch json directly to versioned type json.